managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java

```diff
@@ -233,84 +224,76 @@ public void asyncUpdateCursorInfo(final String ledgerName, final String cursorNa
                 info.getCursorsLedgerId(), info.getMarkDeleteLedgerId(), info.getMarkDeleteEntryId());
 
         String path = prefix + ledgerName + "/" + cursorName;
-        byte[] content = info.toString().getBytes(Encoding);
+        byte[] content = info.toByteArray();
```

Comment:
Yes, the size concern for the text format is very real, but the breaking change would be dangerous in 2 ways: 
 * While a rolling upgrade is happening, an updated broker crashes and topics can end up in a non-updated broker
 * In case there is any issue with the release, we need to be able to roll back to the previous release

In general this problem can be solved in a couple of ways: 
 * Do one release that will understand both formats. Next release will start writing the new format.
 * Use a config switch to chose the format. Then retire the config variable once the feature is widely activated.

For the sake of this PR, I would say to not mix it with format changes, this we need to tackle separately in a controlled way.

In this case, we can simply avoid snapshotting this information into the z-node. In normal behavior the cursor position is appended in binary form into a ledger. 
When doing graceful topic close, as an optimization to save on the number of zk writes to do, we write the information with the last mark-delete position in the z-node and throw the ledger away.

What we could do, when closing the topic, is: 
 * If we have individually-deleted-messages, we close the cursor ledger
 * If not, continue with current behavior of storing snaphot in the z-node

In both cases the information is preserved.

Later, when we enable binary format (btw: we should also do that for managed-ledger z-nodes), we can revert to a unique behavior again.